code/game/machinery/cryopod.dm

```diff
@@ -433,16 +436,20 @@ GLOBAL_VAR_INIT(cryopods_enabled, FALSE)
 	message_admins("[key_name_admin(target)] entered a stasis pod. (<A HREF='?_src_=holder;[HrefToken()];adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)")
 	add_fingerprint(target)
 
+#define JOIN_SLEEP_DURATION 6 SECONDS
+
 /obj/machinery/cryopod/JoinPlayerHere(mob/M, buckle)
 	. = ..()
-	open_machine()
+	close_machine(M, TRUE) // put the mob inside instead of on the turf
+	playsound(src, join_sound, 30)
 	if(iscarbon(M))
 		apply_effects_to_mob(M)
+	addtimer(CALLBACK(src, PROC_REF(open_machine)), JOIN_SLEEP_DURATION)
```

Comment:
what happens if someone else is in this cryo machine?

Reply:
It can't be selected as a spawn location if it's already occupied, and you can't enter it until the joining player gets ejected from it.